controller/internal/enforcer/nfqdatapath/datapath_tcp.go

```diff
@@ -800,18 +805,20 @@ func (d *Datapath) processNetworkAckPacket(context *pucontext.PUContext, conn *c
 		if err := tcpPacket.CheckTCPAuthenticationOption(enforcerconstants.TCPAuthenticationOptionBaseLen); err != nil {
 			// TODO: this needs to be converted to a rejected packet messages. It doesn't mean rejected flow. Disabling.
 			// d.reportRejectedFlow(tcpPacket, conn, collector.DefaultEndPoint, context.ManagementID(), context, collector.InvalidHeader, nil, nil)
-			return nil, nil, fmt.Errorf("TCP authentication option not found: %s", err)
+			return nil, nil, errTCPAuthNotFound
 		}
 
 		if _, err := d.tokenAccessor.ParseAckToken(&conn.Auth, tcpPacket.ReadTCPData()); err != nil {
 			d.reportRejectedFlow(tcpPacket, conn, collector.DefaultEndPoint, context.ManagementID(), context, collector.InvalidToken, nil, nil)
-			return nil, nil, fmt.Errorf("Ack packet dropped because signature validation failed: %s", err)
+			zap.L().Debug("Error: Ack Packet dropped because signature validation failed", zap.Error("error", err))
+			return nil, nil, errSigValidFailed
```

Comment:
don't use a constant error here. parse token can fail due to multiple reasons. need the error report by that function to propagate